packages/block-library/src/html/edit.js

```diff
@@ -5,56 +5,49 @@ import { __ } from '@wordpress/i18n';
 import { Component } from '@wordpress/element';
 import { BlockControls, PlainText, transformStyles } from '@wordpress/editor';
 import { Disabled, SandBox } from '@wordpress/components';
-import { select } from '@wordpress/data';
-
-/**
- * Parse editor styles.
- *
- * @return {Array} css rule set array.
- */
-const getEditorStyles = () => {
-	const { getEditorSettings } = select( 'core/editor' );
-	const styles = getEditorSettings().styles;
-	if ( styles && styles.length > 0 ) {
-		return transformStyles( styles );
-	}
-
-	return [];
-};
+import { withSelect } from '@wordpress/data';
 
 class HTMLEdit extends Component {
 	constructor() {
 		super( ...arguments );
-
 		this.state = {
-			styles: [],
 			isPreview: false,
+			styles: [],
 		};
+		this.switchToHTML = this.switchToHTML.bind( this );
+		this.switchToPreview = this.switchToPreview.bind( this );
 	}
 
 	componentDidMount() {
-		if ( ! HTMLEdit.styles ) {
-			HTMLEdit.styles = getEditorStyles();
-		}
+		const { styles } = this.props;
+		this.setState( { styles: transformStyles( styles ) } );
```

Comment:
@youknowriad 
I am concerned about consumption of memory. 
I think that it is better to use it on all instances of the same computed `styles` .

Reply:
I can see the point here but the problem is that we may have multiple editors on the same page, we can't use a global cache. At best we need a cache per editor. Did you measure the memory consumption, I don't expect it to be very bad but would be good to measure.